src/mbgl/tile/vector_tile.cpp

```diff
@@ -84,6 +84,27 @@ optional<Value> VectorTileFeature::getValue(const std::string& key) const {
     return optional<Value>();
 }
 
+std::unordered_map<std::string,Value> VectorTileFeature::getProperties() const {
+    std::unordered_map<std::string,Value> properties;
+    pbf tags = tags_pbf;
+    while (tags) {
+        uint32_t tag_key = tags.varint();
+        uint32_t tag_val = tags.varint();
+        // this shouldn't use a linear searach of a map
```

Comment:
done in 8bfd1f62f5969de3030cea4e761522da6a6113d2

I stored the original strings in the map and references in the vector because reference_wrapper was causing problems with unordered_map's key requirements. It might be possible to get it to work but this was easier and should work just as well.